### PR TITLE
Review/molgenis data

### PR DESCRIPTION
**Pull request by jjettenn**
#### Checklist
- [ ] Unit tests
- [ ] Updated testplan

### CODE REVIEW

**Comment by bartcharbon** [comment]
test this please

File: molgenis-data/src/main/java/org/molgenis/data/EntityManagerImpl.java [hunk 1]
```diff
@@ -131,7 +131,6 @@ public Entity resolveReferences(EntityType entityType, Entity entity, Fetch fetc
 		// resolve lazy entity collections without references
 		if (entities instanceof EntityCollection && ((EntityCollection) entities).isLazy())
 		{
-			// TODO remove cast after updating DataService/Repository interfaces to return EntityCollections
```
Comment (bartcharbon) [inline molgenis-data/src/main/java/org/molgenis/data/EntityManagerImpl.java]: re-add todo 2 lines higher? above the if

File: molgenis-data/src/main/java/org/molgenis/data/meta/AttributeRepositoryDecorator.java [hunk 2]
```diff
@@ -322,7 +324,200 @@ public void add(Attribute attr)
 	@Override
 	public Integer add(Stream<Attribute> attrs)
 	{
-		return decoratedRepo.add(attrs);
+		return decoratedRepo.add(attrs.filter(attr ->
```
Comment (fdlk) [inline molgenis-data/src/main/java/org/molgenis/data/meta/AttributeRepositoryDecorator.java]: This code has been moved to a separate validation decorator

File: molgenis-data/src/main/java/org/molgenis/data/meta/model/EntityType.java [hunk 3]
```diff
@@ -61,20 +61,6 @@ public EntityType(EntityType entityType)
 		setDefaultValues();
 	}
 
-	/**
```
Comment (fdlk) [inline molgenis-data/src/main/java/org/molgenis/data/meta/model/EntityType.java]: This code is actually called, I believe, when you use the EntityTypeFactory. But IntelliJ cannot detect it cause the factory uses reflection to find the constructor. Let's leave this in place.

**Comment by jjettenn** [comment]
test this please

**Comment by jjettenn** [comment]
test this please

File: molgenis-data/src/main/java/org/molgenis/data/meta/AttributeRepositoryDecorator.java [hunk 4]
```diff
@@ -26,6 +27,7 @@
 import static org.molgenis.security.core.Permission.READ;
 import static org.molgenis.security.core.utils.SecurityUtils.currentUserIsSu;
 import static org.molgenis.security.core.utils.SecurityUtils.currentUserisSystem;
+import static org.molgenis.AttributeType.*;
```
Comment (fdlk) [inline molgenis-data/src/main/java/org/molgenis/data/meta/AttributeRepositoryDecorator.java]: These two imports still need to go I think?

**Pull request merged**